### DISCUSSION TITLE
Is submodule content fetched when git-submodules manager is processing?

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#29249] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

gitlab.com

### Please tell us more about your question or problem

I want a postUpgrade script to run when a git-submodule ref is updated. The script assumes the the content of the submodule was fetched recursively beforehand.
The script is failing with an error that a file from the submodule folder doesn't exist.  Not sure if it is by design. Would appreciate your insights.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Try https://docs.renovatebot.com/configuration-options/#clonesubmodules ?

**Reply by lmilbaum** [reply]
Thanks @rarkins for sharing this option. I wasn't aware of it.  A followup question. Would that also work after the gitsubmodule ref was updated?
I'm asking it because the content of the submodules is used in an after upgrade script.

**Reply by rarkins** [reply]
I'd hope so. If not then we'd need a reproduction using a public submodule example to debug it

**Reply by lmilbaum** [reply]
It didn't work for me. I'm not sure I can generate a public reproduction repo the use case requires a self-hosted runner.
